app/src/main/scala/org/alephium/explorer/service/TransactionService.scala

```diff
@@ -200,4 +217,54 @@ object TransactionService extends TransactionService {
 
   def getTotalNumber()(implicit cache: TransactionCache): Int =
     cache.getMainChainTxnCount()
+
+  def exportTransactionsByAddress(address: Address,
+                                  fromTime: TimeStamp,
+                                  toTime: TimeStamp,
+                                  exportType: ExportType)(
+      implicit ec: ExecutionContext,
+      ac: ActorSystem,
+      dc: DatabaseConfig[PostgresProfile]): Publisher[Buffer] = {
+
+    transactionsPublisher(
+      address,
+      exportType,
+      transactionSource(address, fromTime, toTime)
+    )
```

Comment:
you are ahead of me :) I saw with visualVM that the flow doesn't seem to close the db action. I need to dig this.

Reply:
It seems reactive-stream does not really support cancellation: https://github.com/reactive-streams/reactive-streams-jvm/tree/v1.0.4#1-publisher-code

Reply:
you can cancel the subscription, since no one will request data, the publisher won't publish further. But I need to validate this first. I'll hack a bit the code just to test.

Reply:
The stream is pull-based or push-based? i.e. Subscriber pull data from Publisher, or Publisher push data to Subscriber

Reply:
If no one requests data from the publisher and the db stream is not done, will the publisher be waiting for new subscriptions forever? If so, this might create a memory leak issue.

Reply:
I'll need to experiment a bit to be able to answer you.

Reply:
![maim-1669650507](https://user-images.githubusercontent.com/2979182/204321864-7c36b55b-3cb4-49e6-95f5-7bd88b28e9ff.png)

Here are bit of `visual-vm`. On the first 5minutes we just let the app run without any query. At 4:05PM we start the streaming export of our biggest address. at around 4:17PM the export is done. From then we see that the app is keeping a high amount of memory even when automatically garbage collecting, I don't know if the JVM is playing it safe by keeping some memory available in case such new query happen. At 4:34PM I ran the `Perfrom GC` from `visual-vm`.

Edit. I made it run longer and ran 10 exports of the same above address in parallel (just before 5PM). This time I was more patient and didn't run the `Perform GC`, it seems to clean itself over time 

![maim-1669655053](https://user-images.githubusercontent.com/2979182/204338367-e3fa88a7-b67c-4134-8b7a-3398bea07d35.png)

Reply:
Even if there is a memory leak issue, it cannot be detected by `visual-vm` I think. Each query might only leak hundreds bytes of most likely.

Let's make sure that we could cancel the SQL stream first, then we will have a better picture of this potential memory issue.

Reply:
FYI I have this open issue https://github.com/softwaremill/tapir/issues/2593 where I have some progress, but still I'm currently stuck :tired_face: